lib/pmdtester/builders/rule_set_builder.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'set'
+require_relative '../cmd'
+module PmdTester
+  # This class is responsible for generation dynamic configuration
+  # according to the difference between base and patch branch of Pmd.
+  class RuleSetBuilder
+    ALL_RULE_SETS = Set['bestpractices', 'codestyle', 'design', 'documentation', 'errorprone',
+                        'migrating', 'multithreading', 'performance', 'regex', 'security'].freeze
```

Comment:
Ok, these additionally must match the names of the xml ruleset files for the categories and the package names of the packages, in which the rules are... (in case of Java based rules).
I think, by convention, this works. I just wasn't aware about the (sub-)package name requirement, until I've seen the method `add_rule_set`.

Reply:
I found that the `UnnecessaryCastRule` in migrating directory have not been referenced by any rule sets, so I removed 'migrating'. And I also remove 'regex' since [RegexHelper](https://github.com/pmd/pmd/blob/d6ec50b20f5a24dc8ceb083d792c41cebc930a35/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/regex/RegexHelper.java) in regex directory is not a rule class.